client/lib/cart-values/types.ts

```diff
@@ -1,55 +1,14 @@
 /**
  * Internal dependencies
  */
-import type { GSuiteProductUser } from 'lib/gsuite/new-users';
-import type { DomainContactDetails } from 'my-sites/checkout/composite-checkout/types/backend/domain-contact-details-components';
+import {
+	ResponseCart,
+	ResponseCartProduct,
+	ResponseCartProductExtra,
+} from 'calypso/my-sites/checkout/composite-checkout/hooks/use-shopping-cart-manager/types';
 
-export type CartItemValue = {
-	product_id?: number;
-	product_name?: string;
-	product_slug?: string;
-	meta?: string;
-	cost?: number;
-	orig_cost?: number | null;
-	currency?: string;
-	volume?: number;
-	free_trial?: boolean;
-	is_domain_registration?: boolean;
-	extra?: CartItemExtra;
-};
+export type CartItemValue = ResponseCartProduct;
```

Comment:
What's the significance of moving these from all optional properties to required properties? Is that a safe assumption of would it need to be wrapped in a Partial to make sure the properties aren't assumed to have values?

Reply:
That's an interesting point. It's hard for me to easily know, since we don't have typechecking across Calypso.

However, here's my reasoning:

I added these types in https://github.com/Automattic/wp-calypso/pull/41639 when I had much less understanding of TypeScript. They provided some type safety for the data manipulated by this library. I'm not sure they were much used by anyone else.

When we created the shopping cart manager, we much more exhaustively determined the data structure of the cart for `ResponseCart` and `ResponseCartItem`, and more importantly, we added properties there that were in many cases not also added here.

My goal is to move the shopping cart manager to its own package, taking all its necessary types with it, and also getting rid of the other cart systems in calypso, including this library. So at this point if there's something in calypso that won't work with the types from the shopping cart manager, I'd like to fix it there, rather than having two sources of truth for the same data.